### PR TITLE
Admins can now drag ghosts to mobs to put that ghost in control of that mob

### PR DESCRIPTION
**Pull request by MrsTonedOne**
This functions the same way as editing the ckey in vv, and has the same permission checks
It requires the admin also be a ghost to avoid conflicts with other drag drop code.

### CODE REVIEW

**Comment by WJohn** [comment]
Can this be done from any distance? What if you move around while dragging them? What if they get out of sight?

File: code/modules/admin/admin.dm [hunk 1]
```diff
@@ -862,4 +862,36 @@ proc/kick_clients_in_lobby(var/message, var/kick_only_afk = 0)
...
+		question = "This mob already has a user ([tomob.ckey]) in control of it! "
+	question += "Are you sure you want to place [frommob.name]([frommob.key]) in control of [tomob.name]?"
+
+	var/ask = alert(question, "Place ghost in control of mob?", "Yes", "No")
+	if (ask != "Yes")
+		return 1
+
+	if (tomob.ckey)
+		var/mob/dead/observer/ghost = new/mob/dead/observer(tomob,1)
+		ghost.ckey = tomob.ckey
```
Comment (MrPerson) [inline code/modules/admin/admin.dm]: ghostize()?

**Comment by MrPerson** [comment]
VERY good idea!

**Comment by RemieRichards** [comment]
Interesting concept :+1: 
Worthless to me though (Shitting mouse is still broken so drag and dropping for me is a pain)

**Comment by Miauw62** [comment]
Amazing :+1:

**Comment by KorPhaeron** [comment]
I wish I'd had this 4 years ago 10/10 feature

File: code/modules/admin/admin.dm (hunk 1 above)
Comment (MrStonedOne) [inline code/modules/admin/admin.dm]: Thank you! I knew there was a proc, but I couldn't find it so I just went off of the assume direct control vv drop down

**Comment by MrStonedOne** [comment]
@WJohn it works. You can move while holding down to put them in a mob on the other side of the station. I put a ghost on the arrivals shuttle into poly.

It also still works if they move between you starting the drag and finishing it.

@MrPerson changed the shit.

**Comment by JJRcop** [comment]
What happens if a ghost manages to re-enter corpse mid-drag? Does BYOND support this and cancel the drag, or would you runtime?

File: code/modules/admin/admin.dm [hunk 2]
```diff
@@ -862,4 +862,34 @@ proc/kick_clients_in_lobby(var/message, var/kick_only_afk = 0)
...
+	if (ask != "Yes")
+		return 1
+
+	tomob.ghostize(0)
+
+	message_admins("<span class='adminnotice'>[key_name_admin(usr)] has put [frommob.ckey] in control of [tomob.name].</span>")
+	log_admin("[key_name(usr)] stuffed [frommob.ckey] into [tomob.name].")
+	feedback_add_details("admin_verb","CGD")
+
+	tomob.ckey = frommob.ckey
```
Comment (Miauw62) [inline code/modules/admin/admin.dm]: you probably want to use transfer_to() here.

**Comment by Jordie0608** [comment]
@JJRcop I'm pretty sure MouseDrop is only run when the click is released, in which case it'd fail to pass isobserver(user) if the ghost has returned to a body.

File: code/modules/admin/admin.dm (hunk 2 above)
Comment (MrStonedOne) [inline code/modules/admin/admin.dm]: I thought about that, and the big issue is if the new mob is a special antag, like nuke ops or wizard, we want the ghost to keep the new mob's mind.

Another note, if that ghost was say, an apprentice, most cases we aren't gonna want the ghost to still have his mind spells.

I'll have it do it if the tomob has no mind.

This in general brings up the need to have better admin control over minds for cases like this

Comment (MrStonedOne) [inline code/modules/admin/admin.dm]: Edit, ya, i don't want transfer_to, i did some testing.

Without transfer_to:

If both mobs have no mind, a new one gets made. -expected behavior

If both mobs have a mind, the to mob's mob is now used. -expected behavior

if the from mob has a mind, but the to mob does not, a new one gets made -expected behavior in most cases (spawning special rules for events)

if the from mob has no mind, but the tomob does, the tomob's mind gets used -expected behavior.

With transfer_to:

If both mobs have no mind, a new one gets made. -expected behavior

If both mobs have a mind, the from mob's mob is now used. -not expected behavior in most cases (spawning special roles for events, putting ghosts in afk user's bodies for critical antags)

if the from mob has a mind, but the tomob does not, a new one gets made -not expected behavior in most cases (spawning special roles for events)

if the from mob has no mind, but the tomob does, a new mind would have to get made, and used. -not expected behavior in most cases. (putting ghosts in afk user's bodies for critical antags).

**Comment by MrStonedOne** [comment]
isobserver(usr) applies to the current person dragging, not the person being dragged, that src.

If the ghost no longer exists, it would either fail the key check on src, or just not run.

This proc is for MouseDrop, an internal byond proc, and its a type based override, so if the mob being dragged is not a ghost, it won't trigger.

Anywho, I'm gonna add the same checks thats in atom/dragdrop, but checking that the dragged object still exists isn't one of them, so I'm gonna trust that its not needed, since dragging code hasn't broken without it.

I will add it for both mobs after the question, since they could disappear while waiting

**Pull request merged**